src/optiga/optiga.c

```diff
@@ -0,0 +1,1275 @@
+    if (res) {
+        return res;
+    }
+
+    // Verify metadata tags are setup as expected.
+
+    {
+        const uint8_t check_tags[] = {0xC0, 0xD0, 0xD1, 0xD3, 0xE8};
+        res = _verify_metadata(
+            OID_PLATFORM_BINDING,
+            _platform_binding_metadata,
+            sizeof(_platform_binding_metadata),
+            check_tags,
+            sizeof(check_tags));
+        if (res) {
+            return res;
+        }
+    }
+    {
+        const uint8_t check_tags[] = {0xC0, 0xE1, 0xD0, 0xD1, 0xD3};
+        res = _verify_metadata(
+            OID_AES_SYMKEY,
+            _aes_symkey_metadata,
+            sizeof(_aes_symkey_metadata),
+            check_tags,
+            sizeof(check_tags));
+        if (res) {
+            return res;
+        }
+    }
+    {
+        const uint8_t check_tags[] = {0xC0, 0xE8, 0xD0, 0xD1, 0xD3};
+        res = _verify_metadata(
+            OID_HMAC, _hmac_metadata, sizeof(_hmac_metadata), check_tags, sizeof(check_tags));
+        if (res) {
+            return res;
+        }
+    }
+    {
+        const uint8_t check_tags[] = {0xC0, 0xE8, 0xD0, 0xD1, 0xD3};
+        res = _verify_metadata(
+            OID_ARBITRARY_DATA,
+            _arbitrary_data_metadata,
+            sizeof(_arbitrary_data_metadata),
+            check_tags,
+            sizeof(check_tags));
+        if (res) {
+            return res;
+        }
+    }
+    {
+        const uint8_t check_tags[] = {0xC0, 0xE1, 0xD0, 0xD1, 0xD3};
+        res = _verify_metadata(
+            OID_ATTESTATION,
+            _attestation_metadata,
+            sizeof(_attestation_metadata),
+            check_tags,
+            sizeof(check_tags));
+        if (res) {
+            return res;
+        }
+    }
+    {
+        const uint8_t check_tags[] = {0xC0, 0xD0, 0xD1, 0xD3};
+        res = _verify_metadata(
+            OID_COUNTER,
+            _counter_metadata,
+            sizeof(_counter_metadata),
+            check_tags,
+            sizeof(check_tags));
+        if (res) {
+            return res;
+        }
+    }
+    return 0;
+}
+
+int optiga_setup(const securechip_interface_functions_t* ifs)
+{
+    if (ifs == NULL) {
+        return SC_ERR_IFS;
+    }
+    _ifs = ifs;
+
+    util_log("optiga_setup");
+
+    // A timer is used to provide the OPTIGA library with the ability to schedule work on the main
+    // event loop
+    pal_timer_init();
+
+#if FACTORYSETUP == 1 || FACTORY_DURING_PROD == 1
+    int res = _factory_setup();
+    if (res) {
+        util_log("factory setup failed");
+        return res;
+    }
+#endif
+
+    _util = optiga_util_create(OPTIGA_INSTANCE_ID_0, _optiga_lib_callback, NULL);
+    if (NULL == _util) {
+        return SC_OPTIGA_ERR_CREATE;
+    }
+
+    _crypt = optiga_crypt_create(OPTIGA_INSTANCE_ID_0, _optiga_lib_callback, NULL);
+    if (NULL == _crypt) {
+        return SC_OPTIGA_ERR_CREATE;
+    }
+
+    return _verify_config();
```

Comment:
This causes a noticable delay after boot, before the orientation arrows animation starts. Not sure what to do about it.

Reply:
perhaps we could increase the frequency of the timer that controls the SDK?